### ISSUE TITLE
Error on first responder set to nil on 10.12 Sierra

### ISSUE THREAD

**Issue by sp-erik-podetti** [glfw/glfw#876]
[General] ERROR: Setting <private> as the first responder for window <private>, but it is in a different window ((null))! This would eventually crash when the view is freed. The first responder will be set to nil.
<private>

There is a way to avoid or fix this error? It comes with every example when creating the window. It wasn't occurring on max osx 10.11 (El Capitan)

**Comment by elmindreda** [comment]
@sp-erik-podetti What is the error and how would I go about reproducing it?

**Comment by HoMeBoYErik** [comment]
Just a basic main with default init functions on osx Sierra and you will get the message in the console.
For OpenGL headers I user glLoadGen with pointer_cpp spec

```
`int main(int argc, char* argv[])
{
    // Initialize GLFW
    if(!glfwInit())
    {
        exit(EXIT_FAILURE);
    }

#ifdef __APPLE__
    // Select OpenGL 4.1
    glfwWindowHint(GLFW_CONTEXT_VERSION_MAJOR, 4);
    glfwWindowHint(GLFW_CONTEXT_VERSION_MINOR, 1);
#else
    // Select OpenGL 4.3
    glfwWindowHint(GLFW_CONTEXT_VERSION_MAJOR, 4);
    glfwWindowHint(GLFW_CONTEXT_VERSION_MINOR, 3);
#endif
    glfwWindowHint(GLFW_OPENGL_FORWARD_COMPAT, gl::TRUE_);
    glfwWindowHint(GLFW_OPENGL_PROFILE, GLFW_OPENGL_CORE_PROFILE);
    glfwWindowHint(GLFW_RESIZABLE, gl::FALSE_);
    glfwWindowHint(GLFW_OPENGL_DEBUG_CONTEXT, gl::TRUE_);

    // Open the window, get current resolution and set fullscreen
    std::string title = "My Window -- ";
    GLFWmonitor* monitor = glfwGetPrimaryMonitor();
    const GLFWvidmode* mode = glfwGetVideoMode(monitor);
    glfwWindowHint(GLFW_RED_BITS, mode->redBits);
    glfwWindowHint(GLFW_GREEN_BITS, mode->greenBits);
    glfwWindowHint(GLFW_BLUE_BITS, mode->blueBits);
    glfwWindowHint(GLFW_REFRESH_RATE, mode->refreshRate);
    
    window = glfwCreateWindow(800, 600, title.c_str(), nullptr, nullptr);

    if(!window)
    {
        glfwTerminate();
        exit(EXIT_FAILURE);
    }
    glfwMakeContextCurrent(window);
    glfwSetKeyCallback(window, key_callback);

    // Load the OpenGL functions.
    gl::exts::LoadTest didLoad = gl::sys::LoadFunctions();

    if(!didLoad)
    {
        glfwTerminate();
        exit(EXIT_FAILURE);
    }

    // Initialization
    initializeGL();
    resizeGL(mode->width, mode->height);

    // Enter the main loop
    //mainLoop();

    // Close window and terminate GLFW
    glfwTerminate();
    // Exit program
    exit(EXIT_SUCCESS);
}`
```

**Comment by elmindreda** [comment]
This has been fixed with #883.
